### PR TITLE
Spanner: Row check allow multiple rows to be unordered in check

### PR DESCRIPTION
**Pull request by chemelnucfin**
To help writing tests with multiple rows.

### CODE REVIEW

File: spanner/tests/system/test_system.py [hunk 1]
```diff
@@ -220,6 +220,12 @@ def _assert_timestamp(self, value, nano_value):
             self.assertEqual(value.microsecond * 1000, nano_value.nanosecond)
 
     def _check_row_data(self, row_data, expected=None):
+        try:
+            row_data = sorted(row_data)
+            expected = sorted(expected)
+        except TypeError:
```
Comment (dhermes) [inline spanner/tests/system/test_system.py]: @chemelnucfin When will this fail with a `TypeError`?

I'd be more comfortable having the caller be explicit. So you could have a flag `sort=True` and then the callers which are un-sortable would have to change from the default to `sort=False`

**Comment by chemelnucfin** [comment]
I broke up the function into table and row, so this works better now.

**Comment by tseaver** [comment]
Sorting the rows is actually incorrect:  we expect that the values come back in a particular order in some testcases.

**Comment by chemelnucfin** [comment]
@tseaver PTAL

File: spanner/tests/system/test_system.py [hunk 2]
```diff
@@ -219,20 +219,22 @@ def _assert_timestamp(self, value, nano_value):
         else:
             self.assertEqual(value.microsecond * 1000, nano_value.nanosecond)
 
-    def _check_row_data(self, row_data, expected=None):
+    def _check_table_data(self, table_data, expected=None):
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: This name is wrong: we aren't always checking a whole table. if you must split out the checks for a single row (why?), then call this `_check_rows_data`.

File: spanner/tests/system/test_system.py [hunk 3]
```diff
@@ -219,20 +219,22 @@ def _assert_timestamp(self, value, nano_value):
...
-                    self._assert_timestamp(expected_cell, found_cell)
-                elif isinstance(found_cell, float) and math.isnan(found_cell):
-                    self.assertTrue(math.isnan(expected_cell))
-                else:
-                    self.assertEqual(found_cell, expected_cell)
+        self.assertEqual(len(table_data), len(expected))
+        for row, expected in zip(table_data, expected):
+            self._check_row_data(row, expected)
+
+    def _check_row_data(self, row_data, expected):
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: We need another length check here for the individual rows:  Python's `zip` builtin truncates at the shortest component.

File: spanner/tests/system/test_system.py [hunk 4]
```diff
@@ -492,7 +494,7 @@ def test_batch_insert_then_read_all_datatypes(self):
         snapshot = session.snapshot(read_timestamp=batch.committed)
         rows = list(snapshot.read(
             self.ALL_TYPES_TABLE, self.ALL_TYPES_COLUMNS, self.ALL))
-        self._check_row_data(rows, expected=self.ALL_TYPES_ROWDATA)
+        self._check_table_data(rows, self.ALL_TYPES_ROWDATA)
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: Leave the keyword argument in place here.

File: spanner/tests/system/test_system.py [hunk 5]
```diff
@@ -874,7 +875,7 @@ def test_read_w_index(self):
             self.TABLE, MY_COLUMNS, self.ALL, index='contacts_by_last_name'))
 
         expected = list(reversed(
-            [(row[0], row[2]) for row in self._row_data(ROW_COUNT)]))
+            [[row[0], row[2]] for row in self._row_data(ROW_COUNT)]))
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: FWIW, there is no real difference here, given how the check function works.

File: spanner/tests/system/test_system.py [hunk 6]
```diff
@@ -874,7 +875,7 @@ def test_read_w_index(self):
...
         self._check_row_data(rows, expected)
```
Comment (tseaver) [inline spanner/tests/system/test_system.py]: This needs to be `_check_rows_data`.

**Pull request merged**